src/panels/lovelace/cards/hui-picture-glance-card.ts

```diff
@@ -87,6 +88,37 @@ class HuiPictureGlanceCard extends LitElement implements LovelaceCard {
     this._config = config;
   }
 
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    if (changedProps.has("_config")) {
+      return true;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    if (oldHass && (this._entitiesDialog || this._entitiesToggle)) {
```

Comment:
You can make this code a lot more readable by using the "guard clause" approach:

```ts
if (!oldHass) { return true; }

if (this._entitiesDIialog) {
  …
}

if (this._entitiesToggle) {
  …
}

return false;
```